ssl/statem/statem_lib.c

```diff
@@ -788,6 +788,52 @@ static int ssl_method_error(const SSL *s, const SSL_METHOD *method)
 }
 
 /*
+ * ssl_strict_version_check - Check that the specified `version` is supported by
+ * `SSL *` instance
+ *
+ * @s: The SSL handle for the candidate method
+ * @version: Protocol version to test against
+ *
+ * Returns 1 on success, otherwise 0
+ */
+int ssl_strict_version_check(const SSL *s, int version)
+{
+    const version_info *vent;
+    const version_info *table;
+
+    switch (s->method->version) {
+    default:
+        /* Version should be match method version for non-ANY method */
+        return version_cmp(s, version, s->version) == 0;
+    case TLS_ANY_VERSION:
+        table = tls_version_table;
+        break;
+    case DTLS_ANY_VERSION:
+        table = dtls_version_table;
+        break;
+    }
+
+    for (vent = table; vent->version != 0; ++vent) {
+        int cmp;
+
+        if (vent->smeth != NULL &&
+            ssl_method_error(s, vent->smeth()) == 0) {
+          cmp = version_cmp(s, version, vent->version);
+
+          if (cmp < 0) {
+            continue;
+          }
+
+          /* `version` is either bigger than the rest of the table entries, or
+           * is an exact match. In any case - our search is over now
+           */
+          return cmp == 0;
+        }
```

Comment:
I was expecting something like:

    for (vent = table; vent->version != 0 && version_cmp(s, version, vent->version) >= 0; vent++) {
        if (vent->cmeth != NULL && ssl_method_error(s, vent->cmeth()) == 0)
            return 1;
    }
    return 0;

Note that I use cmeth(), not smeth().